### PR TITLE
BPTL Dashboard:  Add homecollection APIs (Part 2.2)

### PR DESCRIPTION
**Pull request by abhinavjonnada82**
https://github.com/episphere/connect/issues/771
Related PR:
https://github.com/episphere/biospecimen/pull/621

Updated APIs to enhance functionality & fix errors

### CODE REVIEW

**Review by Gbarra9 — APPROVED** [review]

**Review by ramavavi — APPROVED** [review]

**Comment by ramavavi** [comment]
Looks good to me.

File: utils/firestore.js [hunk 1]
```diff
@@ -2036,16 +2036,21 @@ const updateKitAssemblyData = async (data) => {
...
+            return true;
+        } else if (supplySnapShot.docs.length !== 0) {
+            return 'duplicate supplykit id';
+        } else if (collectionSnapShot.docs.length !== 0) {
+            return 'duplicate collection id';
+        }
     } catch (error) {
-        console.error(error);
-        return new Error('Error while looking for Collection ID!', { cause: error })
+        return new Error(error);
```
Comment (jhflorey) [inline utils/firestore.js:2051]: NIT: it will lead to a bit of complexity when tracking logs

**Review by jhflorey — APPROVED** [review]

**Review by jhflorey — APPROVED** [review]

File: utils/biospecimen.js [hunk 2]
```diff
@@ -625,10 +625,10 @@ const biospecimenAPIs = async (req, res) => {
         if(Object.keys(requestData).length === 0 ) return res.status(400).json(getResponseJSON('Request body is empty!', 400));
         try {
             const { addKitAssemblyData } = require('./firestore');
-            await addKitAssemblyData(requestData);
-            return res.status(200).json({message: `Success!`, code:200});
+            const response = await addKitAssemblyData(requestData);
```
Comment (anthonypetersen) [inline utils/biospecimen.js:628]: nice

File: utils/biospecimen.js [hunk 3]
```diff
@@ -643,26 +643,26 @@ const biospecimenAPIs = async (req, res) => {
...
         }
     }
-
     else if (api === 'collectionUniqueness'){
         if( req.method !== 'GET') {
             return res.status(405).json(getResponseJSON('Only GET requests are accepted!', 405));
         }
-        const query = req.query.id.slice(0, -3) + " " + req.query.id.slice(-3); // add space to collection
+        const supplyQuery = req.query.supply;
+        const collectionQuery = req.query.collection.slice(0, -3) + " " + req.query.collection.slice(-3); // add space to collection
```
Comment (anthonypetersen) [inline utils/biospecimen.js:658]: does `req.query.collection` need null checked?

File: utils/biospecimen.js [hunk 4]
```diff
@@ -677,9 +677,9 @@ const biospecimenAPIs = async (req, res) => {
         try {
             const { assignKitToParticipant } = require('./firestore');
             const response = await assignKitToParticipant(requestData);
-            return res.status(200).json({data: response, code:200});
+            return res.status(200).json({ response, code:200 });
```
Comment (anthonypetersen) [inline utils/biospecimen.js:680]: nice job using one consistent format

**Review by anthonypetersen — COMMENTED** [review]
left comments

**Review by anthonypetersen — APPROVED** [review]

**Pull request merged**